pinot-core/src/main/java/org/apache/pinot/core/plan/AggregationPlanNode.java

```diff
@@ -179,20 +181,29 @@ public Operator<IntermediateResultsBlock> buildNonFilteredAggOperator() {
     BaseFilterOperator filterOperator = filterPlanNode.run();
 
     if (canOptimizeFilteredCount(filterOperator, aggregationFunctions)) {
-      return new FastFilteredCountOperator(aggregationFunctions, filterOperator, _indexSegment.getSegmentMetadata());
+      return new FastFilteredCountOperator(aggregationFunctions, filterOperator, _indexSegment.getSegmentMetadata(),
+          _queryContext.isNullHandlingEnabled());
     }
 
-    if (filterOperator.isResultMatchingAll()) {
+    boolean skipNonScanBasedAggregation = false;
+    if (filterOperator.isResultMatchingAll() && !_queryContext.isNullHandlingEnabled()) {
       if (isFitForNonScanBasedPlan(aggregationFunctions, _indexSegment)) {
         DataSource[] dataSources = new DataSource[aggregationFunctions.length];
         for (int i = 0; i < aggregationFunctions.length; i++) {
           List<?> inputExpressions = aggregationFunctions[i].getInputExpressions();
           if (!inputExpressions.isEmpty()) {
             String column = ((ExpressionContext) inputExpressions.get(0)).getIdentifier();
             dataSources[i] = _indexSegment.getDataSource(column);
+            NullValueVectorReader nullValueReader = dataSources[i].getNullValueVector();
```

Comment:
This part is not required because it is within the condition of no null handling

Reply:
I have it here (similar to `DistinctPlanNode`) because some some code (e.g. IS NULL, IS NOT NULL operators) handles nulls even when `nullHandlingEnabled` is set to False. Should we still remove this check?

Reply:
I think so because it is under `filterOperator.isResultMatchingAll()` which already handles `IS NULL` and `IS NOT NULL`